api/encoding/call_option.go

```diff
@@ -26,32 +26,84 @@ package encoding
 // Encoding authors should accept yarpc.CallOptions and convert them to
 // encoding.CallOptions to use with NewOutboundCall. This will keep the
 // API for service authors simple.
-type CallOption struct{ apply func(*OutboundCall) }
+type CallOption struct {
+	opt callOption
+}
+
+type callOption interface {
+	apply(*options)
+}
+
+type options struct {
+	headers         []keyValuePair
+	shardKey        *string
+	routingKey      *string
+	routingDelegate *string
+	responseHeaders *map[string]string
+}
+
+type responseHeadersOptions struct {
```

Comment:
nit: `type responseHeadersOptions map[string]string`.

Reply:
I tried doing this but got various errors like:
```
Invalid receiver type 'responseHeadersOptions' ('responseHeadersOptions' is a pointer type)
```

Not the type has to be `type responseHeadersOptions *map[string]string` (observe the pointer)

Reply:
I don't think the pointer receiver is necessary in the `apply` method.

using the type directly may help from a performance perspective (less objects)